packages/expo-av/src/__stories__/components/AudioModeSelector.ios.tsx

```diff
@@ -0,0 +1,154 @@
+
+  _setMode = (interruptionModeIOS: number) => () =>
+    this.setState(state => ({ modeToSet: { ...state.modeToSet, interruptionModeIOS } }));
+
+  _renderToggle = ({
+    title,
+    disabled,
+    valueName,
+    value,
+  }: {
+    title: string;
+    disabled?: boolean;
+    valueName:
+      | 'interruptionModeIOS'
+      | 'playsInSilentModeIOS'
+      | 'allowsRecordingIOS'
+      | 'staysActiveInBackground';
+    value?: boolean;
+  }) => (
+    <View
+      style={{
+        flexDirection: 'row',
+        alignItems: 'center',
+        justifyContent: 'space-between',
+        paddingVertical: 5,
+        borderBottomWidth: 1.0 / PixelRatio.get(),
+        borderBottomColor: '#cccccc',
+      }}>
+      <Text style={{ flex: 1, fontSize: 16 }}>{title}</Text>
+      <Switch
+        disabled={disabled}
+        value={value !== undefined ? value : Boolean(this.state.modeToSet[valueName])}
+        onValueChange={() =>
+          this.setState(state => ({
+            modeToSet: { ...state.modeToSet, [valueName]: !state.modeToSet[valueName] },
+          }))
+        }
+      />
+    </View>
+  );
+
+  _renderModeSelector = ({
+    title,
+    disabled,
+    value,
+  }: {
+    title: string;
+    disabled?: boolean;
+    value: number;
+  }) => (
+    <Button
+      disabled={disabled}
+      title={`${this.state.modeToSet.interruptionModeIOS === value ? '✓ ' : ''}${title}`}
+      onPress={this._setMode(value)}
+    />
+  );
+
+  render() {
+    return (
+      <View style={{ marginTop: 5 }}>
+        {this._renderToggle({
+          title: 'Plays in silent mode',
+          valueName: 'playsInSilentModeIOS',
+        })}
+        {this._renderToggle({
+          title: 'Allows recording',
+          valueName: 'allowsRecordingIOS',
+          disabled: !this.state.modeToSet.playsInSilentModeIOS,
+          value: !this.state.modeToSet.playsInSilentModeIOS ? false : undefined,
+        })}
+        {this._renderToggle({
+          title: 'Stay active in background',
+          valueName: 'staysActiveInBackground',
+          disabled: !this.state.modeToSet.playsInSilentModeIOS,
+          value: !this.state.modeToSet.playsInSilentModeIOS ? false : undefined,
+        })}
+        {this._renderModeSelector({
+          title: 'Mix with others',
+          value: Audio.INTERRUPTION_MODE_IOS_MIX_WITH_OTHERS,
+        })}
+        {this._renderModeSelector({
+          title: 'Do not mix',
+          value: Audio.INTERRUPTION_MODE_IOS_DO_NOT_MIX,
+        })}
+        {this._renderModeSelector({
+          disabled: this.state.modeToSet.playsInSilentModeIOS === false,
+          title: 'Duck others',
+          value: Audio.INTERRUPTION_MODE_IOS_DUCK_OTHERS,
+        })}
+        <Button
+          title="Apply changes"
+          onPress={this._applyMode}
+          style={{ marginTop: 10 }}
+          disabled={this._modesEqual(this.state.modeToSet, this.state.setMode)}
+        />
+      </View>
+    );
+  }
+}
```

Comment:
This whole components is almost 100% the same as `AudioModeSelected.android.tsx`. Let's merge them and handle the differences on the component level, shall we? :muscle: